main.go

```diff
@@ -115,16 +137,63 @@ func isAppUsingOutdatedBuildpack(app cfclient.App, buildpack *cfclient.Buildpack
 	return timeOfLastBuildpackUpdate.After(timeOfLastAppRestage)
 }
 
-func findOutdatedApps(apps []cfclient.App, buildpacks map[string]cfclient.BuildpackResource) {
+func findOwnersOfApps(apps []cfclient.App) map[string][]cfclient.App {
+	// Mapping of users to the apps.
+	owners := make(map[string][]cfclient.App)
+	for _, app := range apps {
+		// Get the space
+		space, err := app.Space()
```

Comment:
If multiple outdated apps live in the same space, we'll look up the same space again for each app. Same goes for space roles. WDYT about caching both to cut down on api calls?

Reply:
@jmcarp great idea! easy fix :)